src/components/RulesetAutomationModal.tsx

```diff
@@ -1,29 +1,37 @@
-import { useContext } from "react"
-import { ModalContext } from "~/state/ModalContext"
+import { useContext } from "react";
+import { ModalContext } from "~/state/ModalContext";
+import FormType from "./FormType";
 
 const RulesetAutomationModal = () => {
+  const { setCreateModalOpened, setRulesetModalOpened, setEventTrigger } = useContext(ModalContext);
 
-  const { setCreateModalOpened, setRulesetModalOpened } = useContext(ModalContext)
+  const handleClose = () => {
+    setCreateModalOpened(false)
+    setRulesetModalOpened(false)
+    setEventTrigger("") // not neccessary but nice to know state is always up-to-date
```

Comment:
When can state by out of date?

Reply:
If you close the modal, the eventTrigger will still be what you pressed on in the Create modal. If you open the Create modal again and press something else, the eventTrigger will change so it doesn't matter but then state would not be a source of truth since there can be those cases where it doesn't matter but it's not up-to-date.

Reply:
By design, having state that can go stale is a liability. What are some ways you can prevent this? Is there a way to get rid of the event trigger state by passing params? Is there a way to pair it up with modal state?

Reply:
I think a potential possibility can be to check if a modal setter is receiving a specific value, then update any other state values. So for example, if `setIsRulesetModalOpened` setter func is receiving a value of false, then update the `eventTrigger` to be an empty string. I believe this is a way is to avoid stale state.